components/form/FFileUpload.vue

```diff
@@ -0,0 +1,259 @@
+   */
+  color?: string;
+  /**
+   * Button text color
+   */
+  textColor?: string;
+  /**
+   * Label, placed on top of input
+   */
+  label?: string;
+  /**
+   * Disable the field
+   */
+  disabled?: boolean;
+  /**
+   * Text color of the label
+   */
+  labelTextColor?: Color;
+  /**
+   * Text and caret error color
+   */
+  errorColor?: Color;
+  /**
+   * Text color of the hint
+   */
+  hintTextColor?: Color;
+  /**
+   * Hide or not the hint / error message
+   */
+  hideHint?: boolean;
+  /**
+   * A hint to display under the input
+   */
+  hint?: string;
+  /**
+   * Icon, displayed before the hint
+   */
+  hintIcon?: Icon | null;
+  /**
+   * Field name. Used in a form context
+   */
+  name?: string;
+  /**
+   * Event that triggers validation
+   */
+  validationTrigger?: 'input' | 'change' | 'focus' | 'blur';
+  /**
+   * Whether the input should be validated on mount
+   */
+  validateOnMount?: boolean;
+  /**
+   * Message to use as hint when validation fails
+   */
+  errorMessage?: string;
+  /**
+   * Loading state of the input
+   */
+  loading?: boolean;
+}
+
+const props = withDefaults(defineProps<FFileUploadProps>(), {
+  modelValue: null,
+  multiple: false,
+  iconName: 'upload',
+  size: 'medium',
+  buttonText: '',
+  allowedMimeTypes: () => [],
+  maximumSize: 50000,
+  rules: () => [],
+  color: 'neutral--light-3',
+  textColor: 'neutral--dark-4',
+  label: '',
+  labelTextColor: 'neutral--dark-4',
+  errorColor: 'danger',
+  hint: '',
+  hideHint: false,
+  hintIcon: null,
+  hintTextColor: 'neutral--dark-4',
+  disabled: false,
+  name: 'defaultName',
+  validateOnMount: false,
+  validationTrigger: 'change',
+  errorMessage: '',
+});
+
+const underlyingFileInputRef = ref<HTMLInputElement>();
+const fileUploadButtonRef = ref<InstanceType<typeof FButton>>();
+
+const { isValid, hint, value, handleValidation } = useFieldWithValidation(
+  props,
+  {
+    validateOnMount: props?.validateOnMount,
+    rules: [isValidFile],
+  }
+);
+
+const emit = defineEmits<{
+  (name: 'update:modelValue', value: File[] | null | undefined): void;
+  (name: 'change', value: Event): void;
+  (name: 'focus'): void;
+  (name: 'files', value: File[]): void;
+}>();
+
+function isValidFile(value: unknown): boolean {
+  const isValidFormat = mimes(value, props.allowedMimeTypes);
+  const isValidSize = sizeRule(value, { size: props.maximumSize });
+
+  const validFile = isValidFormat && isValidSize;
+  if (validFile) {
+    emit('files', value as File[]);
+  }
+
+  return validFile;
+}
+
+const { handleChange } = useInputEventBindings(
+  handleValidation,
+  props.validationTrigger,
+  emit
+);
+
+const hintTextColor = computed(() =>
+  props.disabled
+    ? 'neutral--dark-1'
+    : isValid.value
+    ? props.hintTextColor
+    : props.errorColor
+);
+
+const iconColor = computed(() =>
+  props.disabled ? 'neutral--dark-1' : props.textColor
+);
+
+defineExpose<{
+  focus: () => void;
+}>({
+  focus,
+});
```

Comment:
i would place all the "define" macros in the same region of the file, so first the `const props = withDefaults(defineProps<FFileUploadProps>()`, then the `defineEmit` then the `defineExpose`